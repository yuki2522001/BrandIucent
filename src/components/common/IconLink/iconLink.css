.icon__item {
  text-decoration: none;
}

.icon__list {
  display: flex;
  justify-content: space-between;
  font-size: 20px;
  width: 200px;
  margin-right: 74px;
}