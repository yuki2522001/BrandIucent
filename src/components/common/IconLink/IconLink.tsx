import React, { memo } from "react";
import { IconProps, IconType } from "@common-types/iconLink";
import Icon from "../Icon/Icon";
import "./iconLink.css";

const IconLink: React.FC<IconProps> = memo(({ iconList }) => {
  const renderIconList = (list: IconType[]) => {
    return list.map((item) => (
      <a
        className="icon__item"
        key={item.key}
        title={item.title}
        href={item.url}
      >
        <Icon iconName={item.iconName} />
      </a>
    ));
  };
  return (
    <div data-testid="icon-link" className="icon__list">
      {renderIconList(iconList)}
    </div>
  );
});

export default IconLink;
