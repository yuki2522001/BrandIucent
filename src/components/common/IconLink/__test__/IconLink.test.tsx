import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import IconLink from "../IconLink";
import ICON_LINK from "@constants/iconLink";

describe("IconLink component", () => {
  test("should render IconLink component", () => {
    const { getByTestId } = render(
        <IconLink iconList={ICON_LINK} />
    );
    expect(getByTestId("icon-link")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
        <IconLink iconList={ICON_LINK} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
