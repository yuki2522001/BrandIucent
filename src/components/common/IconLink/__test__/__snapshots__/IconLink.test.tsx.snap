// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="icon__list"
    data-testid="icon-link"
  >
    <a
      class="icon__item"
      href="https://vi-vn.facebook.com/"
      title="Facebook"
    >
      <i
        class="icon-fb"
        data-testid="icon"
      />
    </a>
    <a
      class="icon__item"
      href="https://twitter.com/"
      title="Twitter"
    >
      <i
        class="icon-twitter"
        data-testid="icon"
      />
    </a>
    <a
      class="icon__item"
      href="https://www.instagram.com/"
      title="Instagram"
    >
      <i
        class="icon-instagram"
        data-testid="icon"
      />
    </a>
  </div>
</DocumentFragment>
`;
